int16 center_x
int16 center_y
int16 is_ok